Add debug tooling to deploy-doc-tests/run

moretea opened the pull request:
Add `-d` and `-s` options to `deploy-doc-tests/run` command

If either of these options is invoked, no test is run.

- `-d`, or `--shell-in-container` starts an interactive docker container with the same 'environment' as where the tests are run.
- `-s`, or `--output-shell-script` does what it's name implies; write a
shell script containing all the steps (in the correct order) to the
standard output.

moretea committed:
Add '-d' and '-s' options to deploy-doc-tests/run command.

If either of these options is invoked, no test is run.

-d, or --shell-in-container starts an interactive docker container with
the same 'environment' as where the tests are run.

-s, or --output-shell-script does what it's name implies; write a
shell script containing all the steps (in the correct order) to the
standard output.

vlal-zz commented on deploy-doc-tests/lib/deploy_doc_test/plan.rb:
```diff
@@ -58,6 +58,20 @@ def missing_env_vars
+        lines << ("#" * 80)
+        lines << "# Phase #{phase_name}"
+        @steps_in_phases[phase_name].each do |step|
+          p step
```
does this print the step as well?

vlal-zz reviewed (commented):
LGTM, just one comment

errordeveloper committed:
Upgrade Scope and Net, bump AMIs

errordeveloper committed:
Delete stale Kubernete manifest

frankscholten reviewed (commented):
How about renaming `run` to `deploydoc`?

tnachen committed:
Add task definition for catalogue-db on aws-ecs (#405)

Commit:
Resolve 'Create Stack' issue and update docs (#409)

* update default instance to t2.medium; update docs

* fixes formatting for sub-headers on ecs-weave-shippable docs page

* fixes incorrect path to cloudformation template

errordeveloper committed:
Add `cfn-signal` based on evidence from docs [1, 2] and Github [3] (#412)

It's been said that this should mitigate some of the flakes we have seen with
`Service arn:aws:ecs:... did not stabilize` errors.

It is cleare that `--resource` should refer to the ASG, but it's still
not clear why the docs suggest using `#!/bin/bash -xe` with `cfn-signal -e $? ...`,
as `$?` will always be zere due to `-e`.

It's also not clear whether we really need `/etc/cfn/{cfn-hup,hooks.d/cfn-auto-reloader}.conf`.

[1]: http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-signal.html
[2]: http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/quickref-ecs.html
[3]: https://github.com/convox/rack/blob/46de9944e7e0078f50f282ac5bdcdc89ceed9b8d/provider/aws/dist/rack.json#L1034

pidster reviewed (changes requested):
Please change 'run' to 'deploydoc' per Frank.

vlal committed:
Integrate healthcheck with deploy docs. Merge swarms (#383)

vlal committed:
Docker swarm migrated to dab format. Added infra helpers. (#417)

moretea committed:
Remove debug print statement

moretea committed:
Rename deploydoc command

moretea committed:
Renamed deploydoc runner; also renamed dir to lose one hyphen

pidster reviewed (changes requested):
There's a merge conflict in `deploy-doc-tests/Dockerfile`.
